### DISCUSSION TITLE
Replacement for getPartialReference to insert entities with foreign keys without loading the foreign entity first

### DISCUSSION THREAD

**Discussion by AnnaDamm** [doctrine/doctrine2#11118] [Support Questions]
Hello,

since the latest version, `getPartialReference` is deprecated.
My usecase currently is this:
* I have a User entity
    * id 1 is "system"
* I have a Changelog entity with a foreign key to User (Cangelog.user)
* when adding changes during code, I create a partial reference with User id 1 and insert it into the database:
    ```php
    $systemUser = $em->createPartialReference(User::class, 1);
    $changelog = (new Changelog())->setUser($systemUser);
    $em->persist($changelog);
    $em->flush();
    ```

Because partial reference is deprecated, the new code would look like this:
```php
$systemUser = $userRepository->find(1);
...
```

But that causes an unnecessary database query, just because i need a User object in the Changelog entity. Just using `->setUser((new User)->setId(1))` would tell doctrine to try to insert a new user entity and end in a duplicate key error.

Is there a real replacement for this usecase?

Thank you for your help!

**Comment by MatTheCat** [comment]
If I understand correctly you should be able to use `EntityManagerInterface::getReference` instead.
